src/NHibernate/Impl/CriteriaImpl.cs

```diff
@@ -994,7 +1009,7 @@ public override string ToString()
 		public System.Type GetRootEntityTypeIfAvailable()
 		{
 			if (persistentClass != null)
-				return persistentClass;
+				return (System.Type)persistentClass;
```

Comment:
Forgotten no more required change: `persistentClass` has been reverted to `System.Type`, the cast is no more needed.